spec/workers/recurring_contribution/cancel_subscription_worker_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+require 'sidekiq/testing'
+Sidekiq::Testing.fake!
+
+RSpec.describe RecurringContribution::CancelSubscriptionWorker do
+  describe '#perform' do
+    context 'when there are expired subscriptions' do
+      before do
+        create(:subscription, :paid, expires_at: Date.current)
+      end
+
+      it 'should call the cancel subscription service' do
+        expect(RecurringContribution::Subscriptions::CancelExpired)
+          .to receive(:process).once
```

Comment:
Check what params are being used here. It should only receive expired subscriptions.